include/highfive/bits/H5Node_traits_misc.hpp

```diff
@@ -104,6 +104,28 @@ inline size_t NodeTraits<Derivate>::getNumberObjects() const {
 }
 
 template <typename Derivate>
+inline std::string NodeTraits<Derivate>::getObjectName(size_t index) const {
+    const ssize_t maxLength = 1023;
+    char buffer[maxLength + 1];
+    ssize_t length =
+        H5Lget_name_by_idx(static_cast<const Derivate*>(this)->getId(), ".",
+                           H5_INDEX_NAME, H5_ITER_INC, index,
+                           buffer, maxLength, H5P_DEFAULT);
+    if (length < 0)
+        HDF5ErrMapper::ToException<GroupException>(
+            "Error accessing object name");
+    if (length <= maxLength)
+        return std::string(buffer);
+    char *bigBuffer = new char[length + 1];
```

Comment:
I mean replacing std::string(buffer) by something like std::string(buffer, length) which will enforce Null terminaison exactly at "length" if HDF5 does not do it properly.